lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,124 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, labels=None, *, fmt="%g", position="edge",
+                  padding=0, shifting=0, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+        You may need to adjust the axis limits to fit the labels.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        labels : array-like, optional
+            A list of label texts, that should be displayed. If not given, the
+            label texts will be the data values formatted with *fmt*.
+
+        fmt : str, default: '%g'
+            A format string for the label.
+
+        position : {'edge', 'center'}, default: 'edge'
+            Position of the label relative to the bar:
+
+            - 'edge': the value is the position of the edge.
+                      (cumulative for stacked bars)
+            - 'center': the value shown will be the length of the bar.
```

Comment:
In the code this looks like it affects both the position and the value printed?  I really don't understand this at all, and the docstring should make it clear what "position" and "value" have to do with one another.

Reply:
Sorry didn't explain well, yes it affects both the label position and the value printed, please refer to the [second example](https://26670-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/lines_bars_and_markers/bar_label_demo.html) for better illustration, as below:
```python
ax.bar_label(p1, position='center')
ax.bar_label(p2, position='center') # print 25 32 34 20 -25
ax.bar_label(p2) # default 'edge', print 45 67 64 55 -52
```
![image](https://user-images.githubusercontent.com/26521816/68779647-f2087b00-066f-11ea-9441-1c56c7bd10ed.png)
how could we describe it better?

Reply:
I see what you are trying to do, but this *is* pretty confusing.  I'm probably not the right person to help you because I'm not sure that I have ever plotted a stacked bar in my life.  
But this is not really what we usually call a "position" in matplotlib, this is really two distinct ways to label a bar, so suggest perhaps:

```python:
label_type: string, default: 'edge'
'edge' : 
Label the end of the bar with the bar's absolute value at the end.  
'center': 
Label the center of the bar, with the length of this segment  (useful for stacked bars, i.e. `link_to_your_example`.)
```

Reply:
I didn’t suggest “bar size” because that’s a bit ambiguous.  Does it include the whole bar down to the baseline or just the segment?

Reply:
Well, it's a bit of terminology. Since we need multiple calls to `bar` I see every block of unique color (what you call segment)  as a separate bar. And these bar***s*** are stacked on top of each other. Also if you say "place the label at the center of the bar" you implicitly adopt that perspective.

I'm open to discussion. We should try to find a consistent and understandable description. IMHO it helps if multiple people contribute their impression/expectation here.

Reply:
Combining suggestions into a slightly more verbose one:
```
label_type : {'edge', 'center'}, default: 'edge'

'edge' :   label placed at the end-point of the bar segment, 
           and the value displayed will be the position of that end-point.

'center' : label placed in the center of the bar segment, 
           and the value displayed will be the length of that segment.
           (useful for stacked bars, i.e. `link_to_your_example`.)
```
does it make sense?

Reply:
:+1: For a standard way of formatting see e.g. the paramter `align` of [bar()](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.bar.html).